realm/realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -329,12 +329,15 @@ public String getPrimaryKeyGetter() {
         return getGetter(primaryKey.getSimpleName().toString());
     }
 
+    /**
+     * Checks if a VariableElement is nullable. In case of PrimaryKey field, such field can be
+     * nullable only if its type could be found in {@link Constants#JAVA_TO_NULLABLE_PRIMARYKEY_TYPES}.
+     *
+     * @return {@code true} if a VariableElement is nullable type, {@code false} otherwise.
+     */
     public boolean isNullable(VariableElement variableElement) {
-        // primary keys cannot be nullable
-        if (hasPrimaryKey()) {
-            if (variableElement.equals(getPrimaryKey())) {
-                return false;
-            }
+        if (hasPrimaryKey() && variableElement.equals(getPrimaryKey())) {
```

Comment:
Is this really useful? I mean all primary key types `String, Long, Integer, Short, Boolean` are nullable because the type is. This check is only useful if the type is somehow restricted when used as a PrimaryKey and the purpose of this PR was to remove this restriction?